src/coldfront_plugin_api/urls.py

```diff
@@ -10,16 +12,75 @@
 
 
 class AllocationViewSet(viewsets.ReadOnlyModelViewSet):
+    """
+    This viewset implements the API to Coldfront's allocation object
+    The API allows filtering allocations by any of Coldfront's allocation model attributes,
+    as well as by Allocation Attributes.
+
+    Filtering by allocation model attributes
+    (i.e by pk, start_date, or even by attributes that span relationships, like project PI email)
+    is possible by using the query parameters directly as Django query filters.
+
+    For example, to filter by the allocation's project PI email: "/api/allocations?project__pi__email=test@bu.edu"
+
+    Filtering using this method requires knowledge of the Allocation model schema and Django's filters
+    Documentation of Coldfront Allocation model: https://coldfront.readthedocs.io/en/latest/apidocs/allocations/
+    Documentation for Django's query syntax: https://docs.djangoproject.com/en/5.0/topics/db/queries/#lookups-that-span-relationships
+
+    To filter by allocation attribute (AA) (i.e Quota Limits CPU),
+    name the query as the AA (case insensitive), prefixed with "attr_"
+
+    For all filters, you can also filter with conditional OR,
```

Comment:
Was the requirement only to filter with `OR`? Is there a reason we don't want to support filtering with `AND`

Reply:
The query is written so that you can filter allocations using conditional AND on two or more attributes. I.e This means it's possible to filter for allocations with a CPU quota of 2 AND RAM quota of 4G. I have updated the docstring to explain this functionality.

Kristi only told me to filter on model and allocation attributes, and I just decided that the OR and AND operations were sufficient so far.

Reply:
> you can filter allocations using conditional AND on two or more attributes. I.e This means it's possible to filter for allocations with a CPU quota of 2 AND RAM quota of 4G. 

Sorry I don't understand this, how would that work? Looking at the query that's being built we are only OR-ing the individual queries. How would the URI look like when we filter for allocation with a "CPU quota of 2 AND RAM quota of 4G" vs "a CPU quota of 2 OR RAM quota of 4G"?

Reply:
Sorry. I needed to be even more specific... 
The filter code is written so that you could perform conditional OR for different *values* of the same attribute 
(CPU quota = 2 OR 4), and to perform conditional AND for different *attributes* (CPU quota = 2 && RAM quota = 8G).
It doesn't make sense to implement conditional AND for values of the same attribute (a quota can't be two values at the same time), and as for conditional OR for different attributes, I simply haven't implemented it because I haven't found it necessary.

An example URL using conditional OR (CPU Quota = 2 OR 4 OR 8):
`api/allocations?attr_CPU Quota=2&attr_CPU Quota=4&attr_CPU Quota=8`

Using conditional AND (CPU Quota = 2 AND RAM Quota )
`api/allocations?attr_CPU Quota=2&attr_RAM Quota=8G`

Using conditional AND with OR (CPU Quota = 2 OR 4, AND Ram Quota = 4G OR 16G)
`api/allocations?attr_CPU Quota=2&attr_CPU Quota=4&attr_RAM Quota=4G&attr_RAM Quota=16G`